File: src/bc-shipping/scheduler/src/com/fabrikam/dronedelivery/deliveryscheduler/scheduler/utils/LocationRandomizer.java
package com.fabrikam.dronedelivery.deliveryscheduler.scheduler.utils;

import java.util.Random;

import com.fabrikam.dronedelivery.deliveryscheduler.scheduler.models.invoker.Location;

public class LocationRandomizer {

	private static Random random = new Random();

	public static Location getRandomLocation() {
		Location location = new Location();
		location.setAltitude(random.nextDouble());
		location.setLatitude(random.nextDouble());
		location.setLongitude(random.nextDouble());

		return location;
	}
}

File: src/shipping/workflow/Fabrikam.Workflow.Service/Utils/LocationRandomizer.cs
// ------------------------------------------------------------
//  Copyright (c) Microsoft Corporation.  All rights reserved.
//  Licensed under the MIT License (MIT). See License.txt in the repo root for license information.
// ------------------------------------------------------------

using System;
using Fabrikam.Workflow.Service.Models;

namespace Fabrikam.Workflow.Service.Utils
{
    static class LocationRandomizer
    {
        private static Random Random = new Random();

        public static Location GetRandomLocation()
        {
            Location location = new Location();
            lock (Random)
            {
                location.Altitude = Random.NextDouble();
                location.Latitude = Random.NextDouble();
                location.Longitude = Random.NextDouble();
            }

            return location;
        }
    }
}
